apps/web/app/(app)/environments/[environmentId]/settings/members/actions.ts

```diff
@@ -0,0 +1,218 @@
+"use server";
+
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { createInviteToken } from "@formbricks/lib/jwt";
+import { AuthenticationError, AuthorizationError, ValidationError } from "@formbricks/errors";
+import {
+  deleteInvite,
+  getInviteToken,
+  inviteUser,
+  resendInvite,
+  updateInvite,
+} from "@formbricks/lib/services/invite";
+import {
+  deleteMembership,
+  getAllMembershipsByUserId,
+  getMembershipByUserId,
+  transferOwnership,
+  updateMembership,
+} from "@formbricks/lib/services/membership";
+import { deleteTeam, updateTeam } from "@formbricks/lib/services/team";
+import { TInviteUpdateInput } from "@formbricks/types/v1/invites";
+import { TMembershipRole, TMembershipUpdateInput } from "@formbricks/types/v1/memberships";
+import { TTeamUpdateInput } from "@formbricks/types/v1/teams";
+import { getServerSession } from "next-auth";
+import { hasTeamAccess, hasTeamAuthority, hasTeamOwnership, isOwner } from "@formbricks/lib/auth";
+import { env } from "@/env.mjs";
+
+export const updateTeamAction = async (teamId: string, data: TTeamUpdateInput) => {
+  return await updateTeam(teamId, data);
+};
+
+export const updateMembershipAction = async (
+  userId: string,
+  teamId: string,
+  data: TMembershipUpdateInput
+) => {
+  const session = await getServerSession(authOptions);
+
+  if (!session) {
+    throw new AuthenticationError("Not authenticated");
```

Comment:
Can we maybe move this code into a separate function so we can solve the authentication check with just one line of code in every server action? :-)

Reply:
Sure, I didn't move this into the auth service `packages/lib/auth.ts` because that service does not have the server session (no next auth). So, maybe there should be a function in the `apps/web` itself